Lib/SharedSet.hpp

```diff
@@ -460,14 +460,8 @@ class SharedSet {
   static unsigned hash(const T* arr, size_t len)
   {
     CALL("SharedSet::hash(T*,size_t)");
-
-    unsigned res=1234567890; //arbitrary number, hopefully won't cause problems:)
-    const T* arre=arr+len;
-    while(arr!=arre) {
-      res=Hash::hash(*arr,res);
-      arr++;
-    }
-    return res;
+    static_assert(is_safely_hashable<T>::value, "T must be safely hashable");
```

Comment:
This originally did something bonkers, but it seems it was only ever actually used with `unsigned`. Stick a `static_assert` here to warn future users of `SharedSet<T>`.